### PR TITLE
MULE-10544: Change ProcessingStragy to use SchedulerService instead of

### PR DESCRIPTION
**Pull request by elrodro83**
flow ThreadingProfile
* Make ProcessingStrategy prototypes

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/construct/Pipeline.java [hunk 1]
```diff
@@ -28,7 +29,9 @@
 
   public List<Processor> getMessageProcessors();
 
-  public void setProcessingStrategy(ProcessingStrategy processingStrategy);
+  public void setProcessingStrategyFactory(ProcessingStrategyFactory processingStrategyFactory);
+
+  public ProcessingStrategyFactory getProcessingStrategyFactory();
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/api/construct/Pipeline.java]: Is this public getter really needed?  Surely the processing strategy shouldn't be used outside of Pipeline, and even if it is then the ProcessingStrategy should be exposed and not the factory.

File: core/src/main/java/org/mule/runtime/core/api/processor/factory/AsynchronousProcessingStrategyFactory.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.processor.factory;
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/api/processor/factory/AsynchronousProcessingStrategyFactory.java]: Don't think this is the best package name as these are not processor factories.  I'd put these in the existing `.strategy.` package or maybe `.strategy.factory`, but I don't think that's necessary really.

File: core/src/main/java/org/mule/runtime/core/api/processor/factory/DefaultFlowProcessingStrategyFactory.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.mule.runtime.core.api.processor.ProcessingStrategy;
+import org.mule.runtime.core.construct.flow.DefaultFlowProcessingStrategy;
+
+
+public class DefaultFlowProcessingStrategyFactory implements ProcessingStrategyFactory {
+
+  @Override
+  public ProcessingStrategy create() {
+    return new DefaultFlowProcessingStrategy();
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/api/processor/factory/DefaultFlowProcessingStrategyFactory.java]: Can we make the strategy implementations inner classes.  They should never be used or instantiated outside the context of their factory.

File: core/src/main/java/org/mule/runtime/core/api/processor/factory/ProcessingStrategyFactory.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.processor.factory;
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/api/processor/factory/ProcessingStrategyFactory.java]: -> `org.mule.runtime.core.api.processor.strategy` (move ProcessingStrategy too in this case) or `org.mule.runtime.core.api.processor`.

File: core/src/main/java/org/mule/runtime/core/api/processor/factory/DefaultFlowProcessingStrategyFactory.java (hunk 3 above)
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/api/processor/factory/DefaultFlowProcessingStrategyFactory.java]: Some test cases are referencing those classes, will try to refactor those cases

File: core/src/main/java/org/mule/runtime/core/api/strategy/factory/NonBlockingProcessingStrategyFactory.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+      processingStrategy.setThreadWaitTimeout(threadWaitTimeout);
+    }
+    if (poolExhaustedAction != null) {
+      processingStrategy.setPoolExhaustedAction(poolExhaustedAction);
+    }
+
+    return processingStrategy;
+  }
+
+  // TODO MULE-10544 use internal executor, do not allow it to be passed form outside
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/api/strategy/factory/NonBlockingProcessingStrategyFactory.java]: Need to take into account the ability to test still though.  Maybe using a TestScheduler if not via this means?

File: core/src/main/resources/META-INF/mule-module.properties [hunk 6]
```diff
@@ -41,6 +42,7 @@ artifact.export.classPackages=org.mule.runtime.core,\
                               org.mule.runtime.core.api.service,\
                               org.mule.runtime.core.api.source,\
                               org.mule.runtime.core.api.store,\
+                              org.mule.runtime.core.api.strategy.factory,\
```
Comment (dfeist) [inline core/src/main/resources/META-INF/mule-module.properties]: Remove.

**Pull request merged**